internal/ethapi/transaction_args.go

```diff
@@ -139,10 +162,31 @@ func (args *TransactionArgs) ToMessage(globalGasCap uint64) types.Message {
 		log.Warn("Caller gas above allowance, capping", "requested", gas, "cap", globalGasCap)
 		gas = globalGasCap
 	}
-	gasPrice := new(big.Int)
+	var gasPrice *big.Int
 	if args.GasPrice != nil {
 		gasPrice = args.GasPrice.ToInt()
 	}
+	var feeCap *big.Int
+	if args.FeeCap != nil {
+		feeCap = args.FeeCap.ToInt()
+	}
+	var tip *big.Int
+	if args.Tip != nil {
+		tip = args.Tip.ToInt()
+	}
+	if gasPrice == nil && feeCap == nil && tip == nil {
+		log.Warn("At least one of gasPrice, feeCap, tip should be set. Defaulting to 0 gas price.")
+		gasPrice = new(big.Int)
+	}
+	if gasPrice != nil && feeCap != nil && tip != nil {
+		log.Warn("Only one of gasPrice, feeCap, tip should be set. Defaulting to feeCap+tip")
+		gasPrice = nil
+	}
```

Comment:
Same here, shouldn't it be `if gasPrice != nil && (feeCap != nil || tip != nil)` ?

Reply:
I believe you are correct here.